ledger/accountdb/accountdb.go

```diff
@@ -192,41 +210,119 @@ type persistedAccountData struct {
 	// The address of the account. In contrasts to maps, having this value explicitly here allows us to use this
 	// data structure in queues directly, without "attaching" the address as the address as the map key.
 	addr basics.Address
-	// The underlaying account data
-	accountData baseAccountData
+	// The underlying account data
+	accountData BaseAccountData
 	// The rowid, when available. If the entry was loaded from the disk, then we have the rowid for it. Entries
 	// that doesn't have rowid ( hence, rowid == 0 ) represent either deleted accounts or non-existing accounts.
 	rowid int64
 	// the round number that is associated with the accountData. This field is needed so that we can maintain a correct
-	// lruAccounts cache. We use it to ensure that the entries on the lruAccounts.accountsList are the latest ones.
+	// LRUAccounts cache. We use it to ensure that the entries on the LRUAccounts.accountsList are the latest ones.
 	// this becomes an issue since while we attempt to write an update to disk, we might be reading an entry and placing
-	// it on the lruAccounts.pendingAccounts; The commitRound doesn't attempt to flush the pending accounts, but rather
-	// just write the latest ( which is correct ) to the lruAccounts.accountsList. later on, during on newBlockImpl, we
+	// it on the LRUAccounts.pendingAccounts; The commitRound doesn't attempt to flush the pending accounts, but rather
+	// just write the latest ( which is correct ) to the LRUAccounts.accountsList. later on, during on newBlockImpl, we
 	// want to ensure that the "real" written value isn't being overridden by the value from the pending accounts.
 	round basics.Round
 }
 
+func (pad persistedAccountData) Addr() basics.Address {
```

Comment:
@algorandskiy possible opportunity for fewer copies / performnace — should we make these pointer receivers and then switch to returning references to persistedAccountData as PersistedAccountData implementations?

Reply:
yes. But when I tried it there is a problem with writing pointers into a cache. Having shared copies break some tests and possibly some non-test functionality